nautobot/docs/development/apps/api/platform-features/prepopulating-data.md

```diff
@@ -0,0 +1,183 @@
+
+- Signal Handlers on `nautobot_database_ready`
+- [Django Onboard Tooling](https://docs.djangoproject.com/en/4.2/howto/initial-data/)
+    - [Django Data Migrations](https://docs.djangoproject.com/en/4.2/topics/migrations/#data-migrations)
+    - [Django Fixtures](https://docs.djangoproject.com/en/4.2/topics/db/fixtures/#fixtures)
+- [Design Builder Jobs](https://docs.nautobot.com/projects/design-builder/en/latest/)
+- Creating data in-place where needed
+
+Some of these methods are explained in further detail at the bottom of this page.
+
+The following table details points that are somewhat easily comparable between the different options.
+
+| Feature                        | Signal Handlers on `nautobot_database_ready`             | Django Data Migrations                      | Django Fixtures                                          | Design Builder Jobs                                      | Creating Data In-place Where Needed                      |
+|--------------------------------|----------------------------------------------------------|---------------------------------------------|----------------------------------------------------------|----------------------------------------------------------|----------------------------------------------------------|
+| Startup Impact                 | Exceptions occur after database migrations               | Exceptions occur during database migrations | N/A                                                      | N/A                                                      | N/A                                                      |
+| Execution                      | Automatic                                                | Automatic                                   | Manual                                                   | Manual                                                   | Automatic                                                |
+| Performance Impact             | Each time migrations are rerun                           | Once when migrations are first run          | Once when fixture is loaded                              | Each time design job is run                              | Each time the data is needed                             |
+| Re-running / Idempotence       | Possible, modified data may be overwritten/cause crashes | Impossible                                  | Possible, modified data may be overwritten/cause crashes | Possible, modified data may be overwritten/cause crashes | Possible, modified data may be overwritten/cause crashes |
+| Custom App Needed              | Yes                                                      | Yes                                         | No                                                       | No                                                       | No                                                       |
```

Comment:
Should `Custom App Needed` be `Yes` under `Creating Data In-place where Needed`?